### PR TITLE
Connect to Post Api

### PR DESCRIPTION
**Pull request by wshe874**
# Description

## Related Issue

Solves #84 

## Type of change

- [x] New feature (enhancement)
- [ ] Bug fix
- [ ] Refactor
- [ ] Breaking change
- [ ] Documentation

## How Has This Been Tested?

- [ ] Automated testing
- [x] Manual testing

# Checklist:

- [ ] Does a similar (open or closed) pull request not already exist?
- [x] Is the pull request _head_ repository a fork repository?
- [x] Is the pull request _compare_ branch a development branch?
- [ ] Is the code documented, particularly in hard-to-understand areas?
- [ ] Does the code build without new warnings?
- [x] Has testing been performed that proves changes are effective and work?
- [x] Has a self- and/or peer-review of the code been performed?
- [ ] Have dependent changes been merged and published in downstream modules?
- [ ] Does all new and existing automated testing pass?
- [ ] Is the person responsible for the repository assigned to the pull request?
- [ ] Is the pull request linked to a project?
- [ ] Is the pull request linked to a milestone?

For more information, refer to the Contributing Guidelines and Code of Conduct links at the bottom of this page.

### CODE REVIEW

File: src/hooks/useCommunities.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export const CommunitiesContextProvider = ({ children }) => {
+  const {
+    data: communities,
+    error,
+    mutate,
+    loading,
+  } = useApi("/communities", {});
+
+  return (
+    <HouseContext.Provider
```
Comment (FriedhelmWS) [inline src/hooks/useCommunities.js]: What is house?

**Review by FriedhelmWS — DISMISSED** [review]
LGTM

**Review by FriedhelmWS — DISMISSED** [review]
LGTM

File: src/components/comment/Comment.jsx [hunk 2]
```diff
@@ -37,7 +36,7 @@ const Comment = ({ comment }) => {
           </IconButton>
 
           <p className="post-numofvotes">
-            {comment.commentupvotes - comment.commentdownvotes}
+            {comment.downVotes - comment.upVotes}
```
Comment (Sparye) [inline src/components/comment/Comment.jsx]: Should it be upVotes - downVotes instead?

File: src/pages/newPost/NewPost.jsx [hunk 3]
```diff
@@ -20,6 +23,43 @@ const styles = {
...
+  );
+  
+  const CreatePostClick = ()=>{
+    postCreateApiCall({
+      data: {
+        title:title,
+        bodyText: text,
+      },
+    }).then(function(success){
+      alert("successfully create a post")
```
Comment (Sparye) [inline src/pages/newPost/NewPost.jsx:48]: alert?

File: src/pages/newPost/NewPost.jsx [hunk 4]
```diff
@@ -20,6 +23,43 @@ const styles = {
...
+      navigate("/homepage",{replace:true})
+    }).catch(function(error){
+      alert("fail to create a post")
```
Comment (Sparye) [inline src/pages/newPost/NewPost.jsx:51]: alert?

**Review by Sparye — DISMISSED** [review]

File: src/pages/newPost/NewPost.jsx (hunk 3 above)
Comment (GlockTian) [reply src/pages/newPost/NewPost.jsx:48]: do you mean change alert to something else?

Comment (Sparye) [reply src/pages/newPost/NewPost.jsx:48]: I don't mind keeping it as alert.

Comment (Sparye) [reply src/pages/newPost/NewPost.jsx:48]: I used Chakra UI useToast() to replace the alerts in my PR

File: src/components/comment/Comment.jsx (hunk 2 above)
Comment (GlockTian) [reply src/components/comment/Comment.jsx]: fixed

**Review by Sparye — APPROVED** [review]
LGTM

**Review by yzhe819 — APPROVED** [review]
LGTM

**Pull request merged**